src/scheduler/job_info.c

```diff
@@ -2183,7 +2183,7 @@ resresv_set_use_queue(queue_info *qinfo)
 
 	if (qinfo->has_hard_limit || qinfo->has_soft_limit || qinfo->has_nodes ||
 	    qinfo->is_ded_queue || qinfo->is_prime_queue || qinfo->is_nonprime_queue ||
-	    qinfo->has_resav_limit || qinfo->resv != NULL)
+	    qinfo->has_resav_limit || qinfo->resv != NULL || (qinfo->partition != NULL))
```

Comment:
This will cause any queue with a partition set to get its own equivalence class.  This isn't quite right.  If multiple queues share the same partition, you'll create multiple equivalence classes where there should only be one.  The right way to do this is to add partition to resresv_set.  If it is NULL, it'll be ignored.  If it is not NULL, it'll be used to break jobs into sets.

Reply:
Bhroam as discussed I have added the changes for equivalences classes as suggested. I have also done unit testing and all the test cases are passing.